src/test/functional/tests/pimPage.spec.ts

```diff
@@ -1,129 +1,186 @@
+  const randomEditedEmployeeName = generateRandomString(3) + "Edited";
+  const randomNameForMultiplyDelete1 = generateRandomString(3);
+  const randomNameForMultipluDelete2 = generateRandomString(3);
 
-    test('Delete mulitiply users', async ({ page }) => {
-      const checkIconGetByName1 = await pimPage.getCheckIconByRandomEmployeeName(randomNameForMultiplyDelete1)
-      const checkIconGetByName2 = await pimPage.getCheckIconByRandomEmployeeName(randomNameForMultipluDelete2)
-      await pimPage.addEmployee(newEmployeeTestData.firstName + randomNameForMultiplyDelete1);
-      await pimPage.navigateToSubPage(SubPage.PIM);
-      await pimPage.addEmployee(newEmployeeTestData.firstName + randomNameForMultipluDelete2);
-      await pimPage.navigateToSubPage(SubPage.PIM);
-      await page.locator(checkIconGetByName1).click()
-      await page.locator(checkIconGetByName2).click()
-      await pimPage.multiplyDelete.click()
-      await pimPage.confirmDeleteButton.click();
-      await expect(page.locator(checkIconGetByName1)).toHaveCount(0)
-      await expect(page.locator(checkIconGetByName2)).toHaveCount(0)
-    })
+  test.beforeEach(async ({ page }) => {
+    loginPage = new LoginPage(page);
+    pimPage = new PimPage(page);
+    await loginPage.initialize();
+    await loginPage.navigateToMainPage();
+    await loginPage.loginUser(
+      adminUserTestData.userName,
+      adminUserTestData.password
+    );
+    await pimPage.navigateToSubPage(SubPage.PIM);
+  });
 
-    test('User should search for employee by name', async ({ page }) => {
-      const tableCells = await tableCellsGotByName(page);
-      await pimPage.searchEmployeeByName(adminUserTestData.userName)
-      await page.waitForLoadState('networkidle')
-      await expect(page.locator(tableCells[0])).toBeVisible();
-      await expect(page.locator(tableCells[1])).not.toBeVisible()
-    })
+  test("Admin user should add employee", async ({ page }) => {
+    await pimPage.addEmployee(
+      newEmployeeTestData.firstName + randomNewEmployeeName
+    );
+    await pimPage.navigateToSubPage(SubPage.PIM);
+    await page.waitForLoadState("networkidle");
+    await expect(
+      pimPage.getCellByRandomNewEmplyeeName(randomNewEmployeeName)
+    ).toBeVisible();
+  });
 
-    test('User should reset search that was done by name', async ({ page }) => {
-      await pimPage.searchEmployeeByName(adminUserTestData.userName)
-      const tableCells = await tableCellsGotByName(page);
-      await page.waitForLoadState('networkidle')
-      await expect.soft(page.locator(tableCells[0])).toBeVisible();
-      await expect.soft(page.locator(tableCells[1])).not.toBeVisible()
-      await pimPage.resetButton.click()
-      await page.waitForLoadState('networkidle')
-      await expect(page.locator(tableCells[0])).toBeVisible();
-      await expect(page.locator(tableCells[1])).toBeVisible()
-    })
+  test("Admin user should edit previousely created employee with random name", async ({
```

Comment:
I think the  title is confusing. Why "random name"?  I think you know already which employee you have been created, so it is not random anymore ;) I suggest to remove word "random"

Reply:
and in general, for all method's name - why you always write get...byRANDOM...? why Random? I know that you generated ranom name, but it is confusing. Of course MINOR comment